test/DynamoCoreWpfTests/RecordedTests.cs

```diff
@@ -48,17 +50,34 @@ public class RecordedUnitTestBase : DynamoViewModelUnitTest
         protected WorkspaceViewModel workspaceViewModel = null;
         protected double tolerance = 1e-6;
         protected double codeBlockPortHeight = Configurations.PortHeightInPixels;
+        protected int DispatcherOpsCounter = 0;
+
+        private void Hooks_OperationPosted(object sender, DispatcherHookEventArgs e)
+        {
+            e.Operation.Task.ContinueWith((t) => Interlocked.Decrement(ref DispatcherOpsCounter));
+            Interlocked.Increment(ref DispatcherOpsCounter);
+        }
 
         public override void Setup()
         {
             base.Setup();
 
+            Dispatcher.CurrentDispatcher.Hooks.OperationPosted += Hooks_OperationPosted;
             // Fixed seed randomizer for predictability.
             randomizer = new System.Random(123456);
         }
 
         public override void Cleanup()
         {
+            Dispatcher.CurrentDispatcher.Hooks.OperationPosted -= Hooks_OperationPosted;
+            DispatcherUtil.DoEventsLoop(() => DispatcherOpsCounter == 0);
+
+            var name = TestContext.CurrentContext.Test.Name;
+            using (var currentProc = Process.GetCurrentProcess())
+            {
+                System.Console.WriteLine($"PID {currentProc.Id} Finished test: {name} with DispatcherOpsCounter = {DispatcherOpsCounter}");
```

Comment:
Can we add the memory logging as well?
Maybe we can create a new class `TestDiagnostics` ? or something to avoid copying this code around?